### ISSUE TITLE
assertion failure in particle.ts for implicit handle connections

### ISSUE THREAD

**Issue by sjmiles** [PolymerLabs/arcs#2947]
If I change this recipe:
```
recipe UdnieStyler
  use as image
  use 'udnie-model' as model
  ImageStyler
    image = image
    model = model
  description `image style transfer using Udnie model`
```
and remove the trivial handle bindings...
```
recipe UdnieStyler
  use as image
  use 'udnie-model' as model
  ImageStyler
  description `image style transfer using Udnie model`
```
it produces many instances of this error:
```
assert-web.ts:14 Uncaught (in promise) Error
    at assert (assert-web.ts:14)
    at Particle.addConnectionName (particle.ts:230)
    at match-free-handles-to-connections.ts:28
    at walker.ts:183
    at Array.forEach (<anonymous>)
    at Object.runUpdateList (walker.ts:182)
    at Object.onResultDone (walker.ts:124)
    at walker.ts:137
    at Array.forEach (<anonymous>)
    at Function.walk (walker.ts:135)
```

**Comment by shaper** [comment]
Is this still an issue? @Cypher1

**Comment by mmandlis** [comment]
I'm pretty sure it is still happening, this is bug in MatchFreeHandlesToConnections strategy, so it only surfaces when full Planner is used.

**Comment by Cypher1** [comment]
As far as I know this bug has not been fixed.

**Comment by shans** [comment]
Fixed by #6197 \o/

**Comment by Cypher1** [comment]
Woo! Thank you!
